core/src/modules/ops/export.ts

```diff
@@ -137,4 +137,42 @@ export namespace ExportOps {
 
     return _exports.length;
   }
+
+  /**
+   * Finds failed exports within a time range and sets them back to pending
+   */
+  export async function retryFailedExports(
+    startDate: Date,
+    endDate: Date,
+    destination?: Destination,
+    saveExports = true
+  ) {
+    const where: WhereOptions<Export> = {
+      state: "failed",
+      createdAt: {
+        [Op.gte]: startDate,
+        [Op.lte]: endDate,
+      },
+    };
+
+    if (destination) where.destinationId = destination.id;
+
+    if (saveExports) {
+      const [count] = await Export.update(
+        {
+          state: "pending",
+          sendAt: new Date(),
+          startedAt: null,
+          errorMessage: null,
+          errorLevel: null,
+          retryCount: 0,
+        },
+        { where }
+      );
```

Comment:
👍 one bulk update

Reply:
We may want to add a LIMIT here - Check out the work done here https://github.com/grouparoo/grouparoo/pull/2840

Reply:
@evantahler after trying to test this it looks like limits on update queries won't have an effect for pg or sqlite

![image](https://user-images.githubusercontent.com/4368928/150862071-3c45dc2f-94ca-424e-a104-5cad392596d1.png)

Since this isn't a recurring thing and will only run when someone wants to retry things explicitly, maybe it's ok?

Reply:
Whoa! https://blog.crunchydata.com/blog/simulating-update-or-delete-with-limit-in-postgres-ctes-to-the-rescue

That's an important fact to know!  I guess it doesn't matter then... feel free to skip it for now (as you say, it's an action not part of the tasks)